contracts/governance/FeeSharingProxy.sol

```diff
@@ -67,13 +75,50 @@ contract FeeSharingProxy is SafeMath96, IFeeSharingProxy {
         //TODO can be also used - function addLiquidity(IERC20Token _reserveToken, uint256 _amount, uint256 _minReturn)
         IERC20(_token).approve(loanToken, amount);
         uint poolTokenAmount = ILoanToken(loanPoolToken).mint(address(this), amount);
-        _writeTokenCheckpoint(loanPoolToken, safe96(poolTokenAmount, "FeeSharingProxy::withdrawFees: pool token amount exceeds 96 bits"));
-
-        lastFeeWithdrawalTime = block.timestamp;
-        
+        if (block.timestamp - lastFeeWithdrawalTime[loanPoolToken] >= FEE_WITHDRAWAL_INTERVAL) {
+            lastFeeWithdrawalTime[loanPoolToken] = block.timestamp;
+            _writeTokenCheckpoint(loanPoolToken, safe96(poolTokenAmount, "FeeSharingProxy::withdrawFees: pool token amount exceeds 96 bits"));
+        }
```

Comment:
this worries me. it makes sense to allow fee withdrawal also between checkpoints, but here we're writing a checkpoint only for the pool tokens received on this transaction. what happens with pool tokens which get received in between checkpoints? I don't see this function remembering the unprocessed amount.

Reply:
You're right, thanks.
Updated test to catch this issue and fixed contract.